src/features/api-keys/lib.ts

```diff
@@ -121,10 +153,15 @@ export const createApiKey = async (
 	const { name, description } = getKeyMetadata(req.body);
 
 	if (!options.name) {
-		options.name = name;
+		options.name = name
+			? name
+			: `Automatically generated ${getKeyType(roleName)} key`;
```

Comment:
This function is the basis for all api key generation endpoints as well (like `/api-key/application/:appId/provisioning`), which is also what the SDK methods call. An explicit call to those endpoints doesn't like it should mark the api keys as "Automatically generated", since there was code outside of the API that explicitly called it for a reason that we can't be sure.

Afaik we agreed to only add this `Automatically generated` placeholder only for API keys generated using `/download` & `/download-config`. Should we switch to an extra argument for defining the placeholder text rather than just on the `roleName`? (Or just an extra optional property(/ies) for the ApiKeyOptions, which though should be destructed and not be propagated into `$createApiKey()`.
PS: Sounds like the whole `KeyMetadata` & `getKeyType` code could just be replaced with a
`{ [roneName: string]: string }` dictionary.

Reply:
Also wasn't the target to use the following text v?
> Name: Automatically generated
Description: 'Automatically generated for an image download' | 'Automatically generated during a configuration file generation'

Reply:
I am reverting back to `provisioning` only as agreed before. I couldn't get your point here:
> An explicit call to those endpoints doesn't like it should mark the api keys as "Automatically generated", since there was code outside of the API that explicitly called it for a reason that we can't be sure.